rnote-ui/src/workspacebrowser/mod.rs

```diff
@@ -417,146 +561,20 @@ impl WorkspaceBrowser {
-
-    pub fn selected_workspace_dir(&self) -> Option<PathBuf> {
-        self.selected_workspace_index().and_then(|i| {
-            self.imp()
-                .workspace_list
-                .item(i)
-                .map(|o| PathBuf::from(o.downcast::<WorkspaceListEntry>().unwrap().dir()))
-        })
-    }
-
-    pub fn set_current_workspace_dir(&self, dir: PathBuf) {
-        let i = self.selected_workspace_index().unwrap_or(0);
-
-        let row = self.imp().workspace_list.remove(i as usize);
-        row.set_dir(dir.to_string_lossy().to_string());
-        self.imp().workspace_list.insert(i as usize, row);
-
-        self.select_workspace_by_index(i);
-    }
-
-    pub fn set_current_workspace_color(&self, color: gdk::RGBA) {
-        let i = self.selected_workspace_index().unwrap_or(0);
-
-        let row = self.imp().workspace_list.remove(i as usize);
-        row.set_color(color);
-        self.imp().workspace_list.insert(i as usize, row);
-
-        self.select_workspace_by_index(i);
-    }
-
-    pub fn set_current_workspace_name(&self, name: String) {
-        let i = self.selected_workspace_index().unwrap_or(0);
-
-        let row = self.imp().workspace_list.remove(i as usize);
-        row.set_name(name);
-        self.imp().workspace_list.insert(i as usize, row);
-
-        self.select_workspace_by_index(i);
-    }
-
-    pub fn current_selected_workspace_row(&self) -> Option<WorkspaceRow> {
-        self.imp()
-            .workspace_listbox
-            .selected_row()
-            .and_then(|row| row.child().map(|w| w.downcast::<WorkspaceRow>().unwrap()))
-    }
-
-    pub fn save_to_settings(&self, settings: &gio::Settings) {
-        if let Err(e) = settings.set("workspace-list", &self.imp().workspace_list) {
-            log::error!("saving `workspace-list` to settings failed with Err {}", e);
-        }
-
-        if let Err(e) = settings.set(
-            "current-workspace-index",
-            &self.selected_workspace_index().unwrap_or(0),
-        ) {
-            log::error!(
-                "saving `current-workspace-index` to settings failed with Err {}",
-                e
-            );
-        }
-    }
-
-    pub fn load_from_settings(&self, settings: &gio::Settings) {
-        let workspace_list = settings.get::<WorkspaceList>("workspace-list");
-        // Be sure to get the index before loading the workspaces, else the setting gets overriden
-        let current_workspace_index = settings.uint("current-workspace-index");
-
-        self.imp().workspace_list.replace_self(workspace_list);
-
-        // current workspace index
-        self.select_workspace_by_index(current_workspace_index);
-    }
-
-    fn setup_dir_actions(&self, _appwindow: &RnoteAppWindow) {
-        self.insert_action_group("workspace_action", Some(&self.imp().workspace_actions));
-
-        self.imp()
-            .workspace_actions
-            .add_action(&workspace_action::create_dir(self));
-    }
+    // setup workspace rows
```

Comment:
this block is another functionality, it does not belong to the file row's setup

Reply:
whoops